Source/Core/Core/NetPlayClient.cpp

```diff
@@ -1035,6 +1056,31 @@ u8 NetPlayClient::LocalWiimoteToInGameWiimote(u8 local_pad)
 	return ingame_pad;
 }
 
+void NetPlayClient::HashRam()
+{
+	std::lock_guard<std::mutex> lk(crit_netplay_client);
+	bool isPlayer = false;
+	for (int i = 0; i < 4; ++i)
+	{
+		if (netplay_client->m_pad_map[i] == netplay_client->m_local_player->pid)
+			isPlayer = true;
+		if (netplay_client->m_wiimote_map[i] == netplay_client->m_local_player->pid)
+			isPlayer = true;
+	}
+
+	if (isPlayer)
+	{
+		bool bWii = SConfig::GetInstance().m_LocalCoreStartupParameter.bWii;
+		u64 hash = XXH64(Memory::GetPointer(bWii ? 0x90000000 : 0x80000000), bWii ? 0x4000000 : 0x2000000, 0);
```

Comment:
1. No need to use GetPointer; just use m_pRAM/m_pEXRAM directly.
2. In Wii mode, you want to hash both m_pRAM and m_pEXRAM.
3. In GC mode, you probably want to hash m_pFakeVMEM.
4. Might as well hash m_pL1Cache as well.
5. All these sizes have names: RAM_SIZE, EXRAM_SIZE, FAKEVMEM_SIZE, L1_CACHE_SIZE.